### PR TITLE
Login with text replacement

### PR DESCRIPTION
**Pull request by cdolm92**
## What is this?

Text replacement now works with login and signup text fields.

### CODE REVIEW

File: Kickstarter-iOS/Views/Controllers/LoginViewController.swift [hunk 1]
```diff
@@ -108,7 +110,7 @@ internal final class LoginViewController: UIViewController {
       .observeForControllerAction()
       .observeValues { [weak self] in
         self?.startResetPasswordViewController()
-      }
+    }
```
Comment (theginbin) [inline Kickstarter-iOS/Views/Controllers/LoginViewController.swift]: think this is an extra line of space

File: Kickstarter-iOS/Views/Controllers/SignupViewController.swift [hunk 2]
```diff
@@ -44,8 +63,11 @@ internal final class SignupViewController: UIViewController, MFMailComposeViewCo
     _ = self.disclaimerButton
       |> disclaimerButtonStyle
 
-    _ = self.emailTextField
-      |> emailFieldStyle
+    _ = self.nameTextField
+      |> UITextField.lens.returnKeyType .~ .next
+
+    _ = self.emailTextField |> emailFieldStyle
+      <> UITextField.lens.returnKeyType .~ .next
```
Comment (theginbin) [inline Kickstarter-iOS/Views/Controllers/SignupViewController.swift]: nice to add the key types! 

you can also set this up like this:
```
 _ = self.emailTextField 
   |> emailFieldStyle
   |> UITextField.lens.returnKeyType .~ .next

File: Kickstarter-iOS/Views/Controllers/SignupViewController.swift [hunk 3]
```diff
@@ -44,8 +63,11 @@ internal final class SignupViewController: UIViewController, MFMailComposeViewCo
     _ = self.disclaimerButton
       |> disclaimerButtonStyle
 
-    _ = self.emailTextField
-      |> emailFieldStyle
+    _ = self.nameTextField
+      |> UITextField.lens.returnKeyType .~ .next
```
Comment (theginbin) [inline Kickstarter-iOS/Views/Controllers/SignupViewController.swift]: there is a style for `nameTextField` on line 75, can you add this lens to that line to condense them?

File: Kickstarter-iOS/Views/Controllers/SignupViewController.swift [hunk 4]
```diff
@@ -56,8 +78,8 @@ internal final class SignupViewController: UIViewController, MFMailComposeViewCo
     _ = self.newsletterLabel
       |> newsletterLabelStyle
 
-    _ = self.passwordTextField
-      |> passwordFieldStyle
+    _ = self.passwordTextField |> passwordFieldStyle
+      <> UITextField.lens.returnKeyType .~ .go
```
Comment (theginbin) [inline Kickstarter-iOS/Views/Controllers/SignupViewController.swift]: same comment as above for putting these on 2 lines with `|>` pipe-forward

File: Kickstarter-iOS/Views/Controllers/LoginViewController.swift [hunk 5]
```diff
@@ -33,9 +33,11 @@ internal final class LoginViewController: UIViewController {
     self.emailTextField.addTarget(self,
                                   action: #selector(emailTextFieldDoneEditing),
                                   for: .editingDidEndOnExit)
+
     self.emailTextField.addTarget(self,
                                   action: #selector(emailTextFieldChanged(_:)),
-                                  for: .editingChanged)
+                                  for: [.editingDidEndOnExit, .editingChanged])
+
```
Comment (theginbin) [inline Kickstarter-iOS/Views/Controllers/LoginViewController.swift]: Curious, have you run this in tests? Wondering if anything would behave differently now that both inputs will be fired on exit.

**Comment by cdolm92** [comment]
@kickstarter/native-squad bitmoji ?

File: Kickstarter-iOS/Tests/Views/DashboardViewControllerTests.swift [hunk 6]
```diff
@@ -10,7 +10,7 @@ internal final class DashboardViewControllerTests: TestCase {
 
     let project = cosmicSurgery
       |> Project.lens.dates.launchedAt .~ (self.dateType.init().timeIntervalSince1970 - 60 * 60 * 24 * 14)
-      |> Project.lens.dates.deadline .~ (self.dateType.init().timeIntervalSince1970 + 1209600)
+      |> Project.lens.dates.deadline .~ (self.dateType.init().timeIntervalSince1970 + 60 * 60 * 24 * 14)
```
Comment (theginbin) [inline Kickstarter-iOS/Tests/Views/DashboardViewControllerTests.swift]: hey you shouldn't have to be fixing this, it's still not passing for you?

Comment (cdolm92) [inline Kickstarter-iOS/Tests/Views/DashboardViewControllerTests.swift]: I changed it back to how it was before.

Comment (theginbin) [inline Kickstarter-iOS/Tests/Views/DashboardViewControllerTests.swift]: oh oh right!

**Comment by mbrandonw** [comment]
nice @cdolm92 ! thanks for taking this one on!

**Pull request merged**